source/common/router/router.cc

```diff
@@ -211,18 +211,24 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   }
 
   // Determine if there is a direct response for the request.
-  if (route_->directResponseEntry()) {
-    auto response_code = route_->directResponseEntry()->responseCode();
-    if (response_code >= Http::Code::MultipleChoices && response_code < Http::Code::BadRequest) {
-      config_.stats_.rq_redirect_.inc();
-      Http::Utility::sendRedirect(*callbacks_, route_->directResponseEntry()->newPath(headers),
-                                  response_code);
-      return Http::FilterHeadersStatus::StopIteration;
-    }
+  const auto* direct_response = route_->directResponseEntry();
+  const auto& request_headers = headers;
```

Comment:
why do you need this variable? Is this just for the capture? can you just capture the headers reference directly?

Reply:
The lambda has its own function parameter named `headers`, so this is just an alias to prevent a name collision.

Reply:
ah OK. FYI you can use C++14 lambda capture expression to define this variable inline in the capture which is a bit cleaner.